### DISCUSSION TITLE
GDScript Interop ?

### DISCUSSION THREAD

**Discussion by eimfach** [migueldeicaza/SwiftGodot#482] [General]
Does it exist and if yes how does it work with existing GDScript Code ?

**Comment by migueldeicaza** [comment]
Given an object, you can invoke a function on it like this:

```swift
func callOn (node: Node) {
    node.call (method: "my_method", Variant (10)
}
```

That would invoke the "my_method" function in GDScript.

**Reply by eimfach** [reply]
I see. So, if I add a children of the "swift node" as a member, their type would always be "Node" ?

**Reply by migueldeicaza** [reply]
It works on any Object, not just Node

**Comment by MrZak-dev** [comment]
What about calling Swift from GDScript , does it work similar to how C# does ? 

can GDScript have auto-completion of SwiftClass' members ?

**Comment by migueldeicaza** [comment]
You can call Swift from GDScript, that is what @Callable is for:

https://migueldeicaza.github.io/SwiftGodotDocs/documentation/swiftgodot/customtypes#Surfacing-Methods

Does that scenario cover what you had in mind?
